server/server.go

```diff
@@ -301,189 +144,213 @@ type Sealer interface {
 	Unseal(*protocol.Operation) ([]byte, error)
 }
 
-type request struct {
-	pkt *protocol.Packet
-	// time just after the request was deserialized from the connection
-	reqBegin time.Time
-	connName string
+type handler struct {
+	name     string
+	s        *Server
+	tokens   chan struct{}
+	mtx      sync.Mutex
+	limited  bool
+	listener net.Listener
+	conn     net.Conn
 }
 
-type response struct {
-	id        uint32
-	op        protocol.Operation
-	reqOpcode protocol.Op
-	err       protocol.Error
-	// time just after the request was deserialized from the connection
-	reqBegin time.Time
+func (h *handler) handle(pkt *protocol.Packet) {
+	var resp response
+	start := time.Now()
+	if h.limited {
+		resp = h.s.limitedDo(pkt, h.name)
+	} else {
+		resp = h.s.unlimitedDo(pkt, h.name)
+	}
+	respPkt := protocol.Packet{
+		Header: protocol.Header{
+			MajorVers: 0x01,
+			MinorVers: 0x00,
+			Length:    resp.op.Bytes(),
+			ID:        resp.id,
+		},
+		Operation: resp.op,
+	}
+	h.tokens <- struct{}{}
```

Comment:
Would be useful to add a comment here or on line 150 where `tokens` is defined, about what it is and how it's used.